google/resource_compute_disk.go

```diff
@@ -390,213 +747,359 @@ func resourceComputeDiskDelete(d *schema.ResourceData, meta interface{}) error {
-			edition = "ent"
-		case "standard":
-			edition = "std"
-		case "express":
-			edition = "exp"
-		}
+	project, err := getProject(d, config)
+	if err != nil {
+		return nil, err
+	}
 
-		var f string
-		if revision := parts[4]; revision != "" {
-			// With revision
-			f = fmt.Sprintf("sql-%s-%s-win-%s-r%s", edition, sqlVersion, windowsVersion, revision)
-		} else {
-			// No revision
-			f = fmt.Sprintf("sql-%s-%s-win-%s", edition, sqlVersion, windowsVersion)
+	// Get the zone
+	z, err := getZone(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	zone, err := config.clientCompute.Zones.Get(project, z).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if v, ok := d.GetOk("image"); ok {
+		log.Printf("[DEBUG] Resolving image name: %s", v.(string))
+		imageUrl, err := resolveImage(config, project, v.(string))
+		if err != nil {
+			return nil, fmt.Errorf(
+				"Error resolving image name '%s': %s",
+				v.(string), err)
 		}
 
-		if f == familyName {
-			return true
+		obj["sourceImage"] = imageUrl
+		log.Printf("[DEBUG] Image name resolved to: %s", imageUrl)
+	}
+
+	if v, ok := d.GetOk("type"); ok {
+		log.Printf("[DEBUG] Loading disk type: %s", v.(string))
+		diskType, err := readDiskType(config, zone, project, v.(string))
+		if err != nil {
+			return nil, fmt.Errorf(
+				"Error loading disk type '%s': %s",
+				v.(string), err)
 		}
+
+		obj["type"] = diskType.SelfLink
 	}
 
-	return false
+	return obj, nil
 }
 
-// e.g. image: windows-server-1709-dc-core-v20180109, family: windows-1709-core
-// e.g. image: windows-server-1709-dc-core-for-containers-v20180109, family: "windows-1709-core-for-containers
-func suppressWindowsFamilyDiff(imageName, familyName string) bool {
-	updatedFamilyString := strings.Replace(familyName, "windows-", "windows-server-", 1)
-	updatedFamilyString = strings.Replace(updatedFamilyString, "-core", "-dc-core", 1)
+func resourceComputeDiskDecoder(d *schema.ResourceData, meta interface{}, res map[string]interface{}) (map[string]interface{}, error) {
+	if v, ok := res["diskEncryptionKey"]; ok {
+		original := v.(map[string]interface{})
+		transformed := make(map[string]interface{})
+		// The raw key won't be returned, so we need to use the original.
+		transformed["raw_key"] = d.Get("disk_encryption_key.0.raw_key")
+		transformed["sha256"] = original["sha256"]
+		if v, ok := d.GetOk("disk_encryption_key_raw"); ok {
+			transformed["raw_key"] = v
+		}
+		d.Set("disk_encryption_key_sha256", original["sha256"])
+		res["diskEncryptionKey"] = transformed
+	}
 
-	if strings.Contains(imageName, updatedFamilyString) {
-		return true
+	if v, ok := res["sourceImageEncryptionKey"]; ok {
+		original := v.(map[string]interface{})
+		transformed := make(map[string]interface{})
+		// The raw key won't be returned, so we need to use the original.
+		transformed["raw_key"] = d.Get("source_image_encryption_key.0.raw_key")
+		transformed["sha256"] = original["sha256"]
+		if v, ok := d.GetOk("disk_encryption_key_raw"); ok {
```

Comment:
There's diskEncryptionKey stuff hanging out here in the sourceImageEncryptionKey block (and below)- intentional?

Reply:
Nope!  Screwed up transitioning from per-attribute decoders (where they were hidden in an if block) to whole-object decoders.  And it didn't trip the unit test, hm.